octavia/common/validate.py

```diff
@@ -440,21 +442,49 @@ def check_cipher_prohibit_list(cipherstring):
     return rejected
 
 
-def check_default_ciphers_prohibit_list_conflict():
+def check_cipher_allow_list(cipherstring):
+    if cipherstring == '' or CONF.api_settings.tls_cipher_allow_list is None:
```

Comment:
cipherstring == '' check seems to be unintended copy, according to the option, only unset (==none) will disable the allow check. Even if unlikely, allow_list='' should disallow all ciphers.

Reply:
The `cipherstring == ''` is the aforementioned fix ported to `check_cipher_allow_list`, but you're right. It isn't needed if I simply check with `not CONF.api_settings.tls_cipher_allow_list`, which I was about to do anyway :)

Reply:
The reason behind having the `cipherstring == ''` check was the aforementioned fix:
> Fix check_cipher_prohibit_list: If cipherstring is empty, return immediately. Else ''.split(':') will be [''], with '' matching the prohibit list, if it is empty, creating a false positive conflict.

1) This made `test_check_default_ciphers_allow_list_conflict` run green before functionality implementation, even though it should run red at that point.
2) If the prohibit list is empty (`''`), Octavia users can provoke a conflict by setting `tls_ciphers` to `:` or something that includes `::`, so that the list resulting from the split includes `''`, which matches the empty prohibit list. This doesn't make sense semantically. However, the error message will tell the user `The following ciphers have been prohibited by an administrator: , ` or something similar.

However, I implemented the check wrong. We only need to return if the prohibit/allow list is empty.
I'll append a fix commit.